elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,217 @@
+     */
+    _isHidden() {
+      return !!this.getAttribute('hidden');
+    }
+
+    /**
+     * Returns a menu item coming after the one passed as an argument, or null.
+     */
+    _nextMenuItem(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (this._isMenuItem(node)) {
+          return next;
+        }
+        next = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns a menu item coming before the one passed as an argument, or null.
+     */
+    _previousMenuItem(node) {
+      let prev = node.previousElementSibling;
+      while (prev) {
+        if (this._isMenuItem(node)) {
+          return prev;
+        }
+        prev = prev.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Opens the menu.
+     */
+    _open() {
+      this._firstMenuItem.setAttribute('tabindex', 0);
+      this._firstMenuItem.focus();
+    }
+
+    /**
+     * Closes the menu.
+     */
+    _close() {
+      this.reset();
+      document.getElementById(
+         this.getAttribute('aria-labelledby')).focus();
+    }
+
+    /**
+     * Makes the element unfocusable, as a reaction to blur event.
+     */
+    _onBlur(event) {
+      if (this._isMenuItem(event.target)) {
+        event.target.setAttribute('tabindex', -1);
+      }
+    }
+
+    /**
+     * Hanldes keyboard interaction.
+     */
+    _onKeyDown(event) {
+      let el = event.target;
+      switch (event.keyCode) {
+        case KEYCODE.DOWN:
+          // If arrow down, move to next item. Wrap if necessary.
+          event.preventDefault();
+          this._moveFocus(el, this._nextMenuItem(el) || this._firstMenuItem);
+          break;
+
+       case KEYCODE.UP:
+         // If arrow up, move to previous item. Wrap if necessary.
+         event.preventDefault();
+         this._moveFocus(el, this._previousMenuItem(el) || this._lastMenuItem);
+         break;
+
+       case KEYCODE.ESCAPE:
+         // If escape, exit the menu.
+         event.preventDefault();
+         this.setAttribute('hidden', true);
+         break;
+
+       case KEYCODE.TAB:
+         this.setAttribute('hidden', true);
+         break;
+
+       default:
+         break;
+      };
+      // If letter key, move to an item which starts with that letter.
+      if (event.keyCode > 64 && event.keyCode < 91) {
+        for (let i = 0, child; child = this.children[i]; i++) {
+          if (child.innerText.trim()[0] === event.key) {
+            event.preventDefault();
+            this._moveFocus(el, child);
+            break;
+          }
+        }
+      }
+    }
+
+    /**
+     * Sets up keyboard interactions for menu and its items.
+     */
+    connectedCallback() {
+      // Make children unfocusable by default.
+      this.reset();
+      this.addEventListener('blur', this._onBlur, true);
+      if (!this.hasAttribute('role')) {
+        this.setAttribute('role', 'menu');
+      }
+      this.setAttribute('hidden', true);
```

Comment:
I think you can remove this and instead do a style like:

`howto-menu:not([opened]) { display: none; }`